<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>548</width>
    <height>679</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>/// Interpreter</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QPushButton" name="runButton">
        <property name="text">
         <string>Run (Ctrl-R)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="checkLimitation">
        <property name="text">
         <string>Limit substitutions to:</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="limitationLineEdit">
        <property name="text">
         <string>1000</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="stepButton">
        <property name="text">
         <string>Step (Ctrl-S)</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="resetButton">
        <property name="text">
         <string>Reset Steps (Ctrl-C)</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="codeBox">
      <property name="title">
       <string>Code</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QTextEdit" name="codeTextEdit"/>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="resultBox">
      <property name="title">
       <string>Result</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QTextEdit" name="resultTextEdit">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="intermedBox">
      <property name="title">
       <string>Intermediate code:</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QTextEdit" name="intermediateTextEdit">
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>checkLimitation</sender>
   <signal>toggled(bool)</signal>
   <receiver>limitationLineEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>111</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>301</x>
     <y>21</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
